snooty/test_parser.py

```diff
@@ -1722,6 +1722,32 @@ def test_definition_list() -> None:
 """,
     )
 
+    # Test a bizarre case
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
```

Comment:
I don't actually know how these tests work, so this is for my own edification. Is this in theory demonstrating that the `collection.createIndex` bit was indented to be definition-list-esque?

Reply:
Yup! This inputs the specific case you ran into, and tests that:
* It doesn't crash
* It doesn't try to create a weird node

I'd like to be a lot stricter about how indentation works and provide a diagnostic, but I don't know what that would look like 😅